drafts/regex-source-generators.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Source Code Generators and regular expressions
+domain: campzulu.hashnode.dev
+tags: architecture, .NET, C#, Blazor
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1672652289997/68d840b3-1d4d-483d-91cd-4a92a638db85.jpeg
+publishAs: linkdotnet
+ignorePost: true
+hideFromHashnodeCommunity: false
+---
+
+
+## Source Code Generators and Regular Expressions
+Since .NET 7 we have the possibility to use source code generators for regular expressions. The first question is: *What is a source code generator?*
+I can quote here [the official Microsoft page](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview):
+> ... Source Generators let C# developers inspect user code as it is being compiled. The generator can create new C# source files on the fly that are added to the user's compilation. In this way, you have code that runs during compilation. It inspects your program to produce additional source files that are compiled together with the rest of your code.
+
+In simple terms: Source code generators hook into the build process and can create additional stuff. And that is where our journey starts with regular expressions. If we define a regular expression, that regular expression can be expressed as code. This happens when you hit the build button.
+
+The advantage over the traditional approach is that we can get the same performance as `new Regex("...", RegexOptions.Compiled)` and the startup benefit of `Regex.CompileToAssembly`, but without the complexity of `CompileToAssembly`. As the code is generated it can be viewed and debugged.
```

Comment:
_but without the complexity of_ I haven't worked with Regex in C# for a while, so when reading this, I am wondering, what is this complexity you refer to? From my dummy perspective, I just see one flag less below :-) Is this complexity well-known for .NET devs? Maybe some context would be nice also for people how are not familiar with the problem :-)

Reply:
Explained it a bit better